docs/markdown/Using-with-Visual-Studio.md

```diff
@@ -14,3 +14,15 @@ In order to generate Visual Studio projects, Meson needs to know the settings of
 If you wish to use the Ninja backend instead of vs2015, pass `--backend ninja`. At the time of writing the Ninja backend is more mature than the VS backend so you might want to use it for serious work.
 
 This assumes Python3 is in your `PATH`, which is highly recommended.
+
+# Using Clang-CL with Visual Studio
+
+You will first need to get a copy of llvm+clang for Windows, such versions are available from a number of sources, including the llvm website. Then you will need the [llvm toolset extension for visual studio](https://marketplace.visualstudio.com/items?itemName=LLVMExtensions.llvm-toolchain). You then need to either use a native file or `set CC=clang-cl`, and `set CXX=clang-cl` to use those compilers, meson will do the rest.
+
```

Comment:
Probably should say something more about 'Use a native file to specify the compiler'?
Also wrap lines?

Reply:
Added a link to the native files documentation and wrapped the whole thing.